### PR TITLE
Speeding up IIS test, organization changes, add HttpsTestFixture.

### PR DESCRIPTION
**Pull request by jkotalik**
This PR improves tests across IIS in many ways:

- Adds Https fixture for all tests that need HTTPS
- Removes always failing test with x86 as it always fails and isn't critical. We can fix it but didn't want to take the effort. I verified it worked locally for sanity as I have an x86 installation of dotnet.
- Delete some redundant stress tests
- Removed some redundant test variations
- Reorganized files into consolidated structures, as the current structure was very unorganized
    - All Http fixture tests are in RequestResponseTests
    - Startup/ShutdownTests now contain most tests.
    - Http2Tests and HttpsTests contain all Https fixture tests (as they are more distinct).

### CODE REVIEW

**Comment by pranavkm** [comment]
> Removes always failing test with x86 as it always fails and isn't critical

Do we get CTI coverage for this?

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/CompressionTests.cs [hunk 1]
```diff
@@ -1,64 +1,96 @@
...
+        public CompressionTests(IISCompressionSiteFixture fixture): base(fixture)
         {
+            _fixture = fixture;
         }
 
-        [ConditionalFact]
-        public async Task PassesThroughCompressionOutOfProcess()
+        [ConditionalTheory]
+        [RequiresIIS(IISCapability.DynamicCompression)]
+        [InlineData(true)]
```
Comment (pranavkm) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/CompressionTests.cs:27]: This is a strong personal preference, but these are much better authored as two separate tests. You could use a private method to share the implementation. Having separate tests makes it much more legible what the test is about, easier to individually skip \ quarantine, and easy to search for \ filter. Theory is much better suited to cover different data to the same algorithm, but is crummy when describing different code paths \ scenarios.

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/ShutdownTests.cs [hunk 2]
```diff
@@ -211,7 +253,6 @@ public async Task AppOfflineDropped_CanRemoveAppOfflineAfterAddingAndSiteWorks(H
         [ConditionalTheory]
         [InlineData(HostingModel.InProcess)]
         [InlineData(HostingModel.OutOfProcess)]
-        [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/7075")]
```
Comment (pranavkm) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/ShutdownTests.cs:214]: Missed unquaranting?

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/ShutdownTests.cs [hunk 3]
```diff
@@ -252,5 +293,136 @@ public async Task AppOfflineAddedAndRemovedStress(HostingModel hostingModel)
...
+            var deploymentResult = await DeployAsync(Fixture.GetBaseDeploymentParameters(hostingModel));
+
+            await deploymentResult.AssertStarts();
+            var load = Helpers.StressLoad(deploymentResult.HttpClient, "/HelloWorld", response =>
+            {
+                var statusCode = (int)response.StatusCode;
+                Assert.True(statusCode == 200 || statusCode == 503, "Status code was " + statusCode);
+            });
+
+            for (int i = 0; i < 100; i++)
```
Comment (pranavkm) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/ShutdownTests.cs:365]:
```suggestion
            for (var i = 0; i < 100; i++)
```

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/ShutdownTests.cs (hunk 2 above)
Comment (jkotalik) [reply src/Servers/IIS/IIS/test/Common.FunctionalTests/ShutdownTests.cs:214]: Yeah, and I did do work on the test in the pass.

**Comment by jkotalik** [comment]
Already seeing big gains with just this PR. Before and after time for the IIS.FunctionalTests.
```
Test Run Successful.
Total tests: 344
     Passed: 303
    Skipped: 41
 Total time: 17.7561 Minutes

Test Run Successful.
Total tests: 304
     Passed: 265
    Skipped: 39
 Total time: 12.5874 Minutes
```

**Review by BrennanConroy — APPROVED** [review]

**Pull request merged**